File: Interview/2023·7·17青训营/2.cpp
class Solution {
public:
    int maxSubArray(std::vector<int>& nums) {
        size_t len = nums.size();
        int sum = 0;
        int ans = 0;

        sum += nums.at(0);
        for (int i = 0; i < len; ++i) {
            if (sum < 0) {
                sum = nums.at(i);
            } else {
                sum += nums.at(i);
            }
            ans = std::max(ans, sum);
        }
        return ans;
    }
};

File: Interview/2023·10·12高途/2.js
function calculate(costs) {
    const size = costs.length
    let index = 0
    let sum = 0
    let ans = 0

    while (index < size) {
        if (sum < 0) {
            sum = 0
        }

        sum += costs[index]
        ans = ans > sum ? ans : sum
        index += 1
    }

    return ans
}
